src/GitTools.jl

```diff
@@ -66,35 +66,24 @@ function transfer_progress(progress::Ptr{LibGit2.TransferProgress}, p::Any)
 end
 
 const GIT_REGEX =
-    r"^(?:(?:git|ssh|https)(?:://|@))?(?:[\w\.\+\-:]+@)?([^/:]+)[:/](.+?)(?:\.git)?$"i
-const GITHUB_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
-const GITLAB_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
-const FALLBACK_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
+    r"^(?:(?<proto>git|ssh|https)://)?(?:[\w\.\+\-:]+@)?(?<hostname>.+?)(?(<proto>)/|:)(?<path>.+?)(?:\.git)?$"
+const GIT_PROTOCOL = Dict{AbstractString, Union{Nothing, AbstractString}}()
 
-function setprotocol!(proto::Union{Nothing, AbstractString}=nothing, site::Symbol=:github)
-    if site == :github
-        GITHUB_PROTOCOL[] = proto
-    elseif site == :gitlab
-        GITLAB_PROTOCOL[] = proto
-    else
-        FALLBACK_PROTOCOL[] = proto
-    end
+function setprotocol!(
+    proto::Union{Nothing, AbstractString}=nothing,
```

Comment:
Is the idea here that `setprotocol!()` unsets the protocol? It's a little weird that it defaults to unsetting the protocol for `github.com`. It seems like `setprotocol!(hostname, [protocol])` might be more natural then? That way omitting the protocol to unset it works the same for every hostname. Also, given that all this does is stick a value into a dict, maybe just expose the dict directly?

Reply:
I definitely like the `setprotocol!(hostname, [protocol])` signature better. I feel like keeping `GitTools.setprotocol!` is a bit neater than just exposing the dict.